src/hotspot/os/bsd/os_bsd.cpp

```diff
@@ -1702,7 +1702,10 @@ bool os::remove_stack_guard_pages(char* addr, size_t size) {
 static char* anon_mmap(char* requested_addr, size_t bytes, bool exec) {
   // MAP_FIXED is intentionally left out, to leave existing mappings intact.
   const int flags = MAP_PRIVATE | MAP_NORESERVE | MAP_ANONYMOUS
-      MACOS_ONLY(| (exec ? MAP_JIT : 0));
+      MACOS_ONLY(| (exec ? MAP_JIT : 0))
+      // On macOS aarch64 MAP_JIT is required if we want to commit an executable chunk
+      // later, so always reserve executable memory right from the start
+      MACOS_AARCH64_ONLY(| MAP_JIT);
```

Comment:
Is this the right way to do it? Where do we ever map memory we want to use for code generation, but not ask for exec permission? Is that a bug?

Reply:
Inside the hotspot VM code we seem to be doing the right thing, however, in the test, i.e. `test/hotspot/gtest/runtime/test_os.cpp` we ask for regular memory, then try to commit a chunk with elevated exec privileges, which seems to work on all other platforms, except aarch64 macOS.

The alternative, would be to fix the test in question, but since this scenario works fine on all the other platforms, I figured the proposed way to handle it in the VM  would be the preferable way to fix it, so future test writers do not need to know this particular platform behavior difference and risking getting it wrong again.